pkg/lib/stampfile-plugin.js

```diff
@@ -0,0 +1,44 @@
+const fs = require('fs');
+const assert = require('assert');
+
+module.exports = class {
+    apply(compiler) {
+        compiler.hooks.afterEmit.tap('StampfilePlugin', compilation => {
+            const stamps = { };
+
+            for (const output in compilation.assets) {
+                const path = compilation.assets[output].existsAt;
+
+                /* `output` will be like
+                 *
+                 *   base1/cockpit.js
+                 *
+                 * `path` will be like
+                 *
+                 *   /build/dir/dist/base1/cockpit.js
+                 *
+                 * We want to get the full absolute path, up to and
+                 * including the first directory component of `output`.
+                 */
+                assert(path.startsWith("/"));
+                assert(path.endsWith(output));
+
+                const first_slash = output.indexOf('/');
+                const dir_length = path.length - output.length + first_slash;
+                const dir = path.slice(0, dir_length);
+
+                /* As per the above example, `dir` would be
+                 *
+                 *   /build/dir/dist/base1
+                 *
+                 * Let's put a stamp there.
+                 */
+                stamps[dir + '/stamp'] = true;
+            }
+
+            for (const stampfile in stamps) {
+                fs.writeFileSync(stampfile, '');
+            }
```

Comment:
Wouldn't it be simpler to directly call writeFileSync() in the first loop, and drop the `stamps` object?

Reply:
The first loop is going to encounter very many files within this subdirectory, so we'd end up writing the stamp file over and over again...

Reply:
The comment about the manifest.json acting as a stamp is reasonable...  we used to use the Makefile.deps for this purpose, in fact...

Reply:
For the record about "where this is going": the idea is eventually that typing `webpack` will produce exactly the same output in `dist/` as if we typed `make`.

Reply:
Loop> make sense, sorry for misreading that.

That's a good goal -- I rather meant whether the stamp file plugin is meant to be a temporary thing or a lasting (part of) the final solution.